libs/sdk-model/src/insight/index.ts

```diff
@@ -618,6 +620,33 @@ export function insightModifyItems<T extends IInsightDefinition>(
     } as T;
 }
 
+/**
+ * Creates a new insight with reduced bucket items (retrieved by applying the modifications function).
+ *
+ * Note: the bucket item modification function SHOULD NOT modify bucket item's localId.
+ * The localId MAY be used to reference the item from other places in the insight (for example from sorts).
+ * Changing the item localId has potential to break the insight: as-is this function does not concern itself with changing the references.
+ *
+ * @param insight - insight to use as template for the new insight
+ * @param modifications - modifications to apply to the bucket items
+ * @returns always new instance
+ * @public
+ */
+export function insightReduceItems<T extends IInsightDefinition>(
+    insight: T,
+    modifications: BucketItemReductions = identity,
+): T {
+    invariant(insight, "insight must be specified");
+    const buckets: IBucket[] = insightBuckets(insight);
+    // tslint:disable-next-line: no-object-literal-type-assertion
```

Comment:
hmm, the project is set with eslint, why the tslint annotation?

Reply:
Done